src/app/app.spec.js

```diff
@@ -1,62 +1,65 @@
 // all of our Jasmine and Sinon utilities (describe, it, sinon) are globally
 // injected by Karma. no need to import them manually for each test.
 import App from './app';
+import localStorageManager from './services/localStorageManagerService/localStorageManager.services';
 
 describe('Application Tests', () => {
-  let sandbox, stateStub, controller, scope, localStorageManagerStub, $timeout;
+  let sandbox, stateStub, $timeout, localStorageManagerStub, controller;
 
   before(() => {
     sandbox = sinon.sandbox.create();
   });
 
   beforeEach(angular.mock.module(App.name));
 
-  beforeEach(inject(($controller, $q, $rootScope, _$timeout_) => {
-    scope = $rootScope.$new();
-    localStorageManagerStub = {getDataFor: sinon.stub()};
+  beforeEach(angular.mock.module(localStorageManager.name));
+
+  beforeEach(() => {
     stateStub = {$state: {go: sandbox.stub()}};
+    localStorageManagerStub = {getDataFor: sinon.stub()};
+  });
+
+  afterEach(() => {
+    sandbox.reset();
+  });
+
+  const createAppController = inject(($controller, _$timeout_, $httpBackend) => {
+    $httpBackend.whenGET('assets/locales/en.json').respond({});
+    $timeout = _$timeout_;
 
     controller = $controller('AppController', {
       $state: stateStub.$state,
       localStorageManager: localStorageManagerStub
     });
-    $timeout = _$timeout_;
-  }));
+  });
 
   afterEach(() => {
     sandbox.resetHistory();
   });
 
   describe('App Controller', () => {
-
     it('should be defined', () => {
+      createAppController();
+
       expect(controller).to.be.ok;
     });
 
-    it('should transition to language selection component', () => {
-      controller.goToLanguageSelection();
-      expect(stateStub.$state.go).to.be.calledOnce;
-      expect(stateStub.$state.go).to.have.been.calledWith('idioma');
-    });
+    it('should decide to go to language when no language found', () => {
```

Comment:
Me quitaría estos decide, 'should go to language when....

Reply:
Visto ahora yo también lo veo mejor sin ellos, los quito.